### PR TITLE
refactor(docs): Updates Environment Variable docs

### LINKED ISSUE #297: Update Environment Variable Guide
There should be detailed instructions on how to setup MyMLH authentication for hackathons. 

_These are separate instructions than the MyMLH setup for development_

### PR DESCRIPTION
**Pull request by cbaudouinjr**
- Adds a better description for environment variables
- Adds a way to contact us for help
- Separates required/recommended variables 
- Updates MyMLH verbiage

# Before
image
image

# After
image
image

closes #297

### CODE REVIEW

File: docs/deployment-environment-variables.md [hunk 1]
```diff
@@ -3,26 +3,33 @@ id: deployment-environment-variables
 title: Environment Variables
 ---
 
-Various services require environment variables to operate.
+Various services require environment variables to operate. While they may appear overwheling, this setup guide aims to alleviate some of the concern. It is recommended you start with the required variables and work through this guide so all are properly setup.
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:6]:
```suggestion
Various services require environment variables to operate. While they may appear overwhelming, this setup guide aims to alleviate some of the concern. 
```
I feel like the required/optional parts below are well laid out enough that we don't need to emphasize that

File: docs/deployment-environment-variables.md [hunk 2]
```diff
@@ -3,26 +3,33 @@ id: deployment-environment-variables
...
+## Recommended
+```bash
+MLH_KEY="my-mlh-application-id"
+MLH_SECRET="my-mlh-secret"
+ROLLBAR_ACCESS_TOKEN="<server-side rollbar token>"
+```
+
 _Also see [app.json](https://github.com/codeRIT/hackathon_manager/blob/master/app.json)_
 
 ### Secret keys
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:35]: These should be up by `## Required` if they are required to run?

File: docs/deployment-environment-variables.md [hunk 3]
```diff
@@ -3,26 +3,33 @@ id: deployment-environment-variables
...
 AWS_REGION="us-east-1"
-ROLLBAR_ACCESS_TOKEN="<server-side rollbar token>"
 SPARKPOST_API_KEY="<sparkpost api key>"
 SPARKPOST_CAMPAIGN_ID="my-hackathon"
 TIME_ZONE="America/New_York"
 ```
 
+## Recommended
+```bash
+MLH_KEY="my-mlh-application-id"
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:28]: Not sure why this code block is here; seems more appropriate to move it under the `MyMLH` section below?

**Review by peterkos — CHANGES_REQUESTED** [review]

**Comment by cbaudouinjr** [comment]
Redesigned the Environment Variable page to reduce the large code blocks. Page is now designed to force the reader to read through the whole page. I also reorganized the keys based on required/optional so they'll show that way on the page navigation.

File: docs/deployment-environment-variables.md [hunk 4]
```diff
@@ -90,37 +84,43 @@ SMTP_AUTHENTICATION="" # optional, "plain" by default
...
 
 ```bash
-ROLLBAR_ACCESS_TOKEN=""
+TIME_ZONE="America/Los_Angeles"
 ```
 
-### My MLH
+By default, charts & timestamps will be in UTC.
+
+## Optional
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:99]: Maybe add `The following environment variables are optional` just to be consistent w/ the description under `##Required`?

File: docs/deployment-environment-variables.md [hunk 5]
```diff
@@ -3,29 +3,20 @@ id: deployment-environment-variables
...
-ROLLBAR_ACCESS_TOKEN="<server-side rollbar token>"
-SENDGRID_API_KEY="<sendgrid api key>"
-TIME_ZONE="America/New_York"
 ```
 
-_Also see [app.json](https://github.com/codeRIT/hackathon_manager/blob/master/app.json)_
-
-### Secret keys
-
 `SECRET_KEY_BASE` and `DEVISE_SECRET_KEY` are required for the app to run. You can generate secrets via `bundle exec rake secret`. This "secret" is a 64-byte hexadecimal string (128 characters). You could also generate this with `head -c 64 /dev/urandom | xxd -ps -c 128` if you are on a standard Linux distribution.
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:20]:
```suggestion
You can generate secrets via `bundle exec rake secret`. This "secret" is a 64-byte hexadecimal string (128 characters). You could also generate this with `head -c 64 /dev/urandom | xxd -ps -c 128` if you are on a standard Linux distribution.
```

Just to alleviate the redundancy as this is stated directly above this env var

File: docs/deployment-environment-variables.md [hunk 6]
```diff
@@ -35,9 +26,9 @@ HM_DOMAIN_NAME=""
 HM_DOMAIN_PROTOCOL=""  # optional, https by default
 ```
 
-### Resumes and S3
+This is needed to allow HackathonManager to send email on behalf of your hackathon's domain.
 
-Resumes are stored locally in development and on S3 in production.
+### Amazon S3 and Resumes
```
Comment (peterkos) [inline docs/deployment-environment-variables.md:31]: I appreciate adding "Amazon S3" vs just "S3"!

**Review by peterkos — CHANGES_REQUESTED** [review]

**Review by peterkos — APPROVED** [review]
:shipit:

**Pull request merged**